homeassistant/components/recorder/models.py

```diff
@@ -117,7 +117,7 @@ class States(Base):  # type: ignore
     last_updated = Column(DateTime(timezone=True), default=dt_util.utcnow, index=True)
     created = Column(DateTime(timezone=True), default=dt_util.utcnow)
     old_state_id = Column(
-        Integer, ForeignKey("states.state_id", ondelete="SET NULL"), index=True
+        Integer, ForeignKey("states.state_id", ondelete="NO ACTION"), index=True
```

Comment:
I think we would need to do a schema migration or am I missing something?
Coping `_update_states_table_with_foreign_key_options(engine)` from version `11` might do the trick.

Reply:
I was trying to avoid migrating the schema since it won't cause any issue for the dbs that support it to leave it as-is.

Reply:
Ok. Does it make sense to add a note (maybe as breaking-change?) that everyone who is using Ms SQL needs to delete their current (none-working) database for the fix to work? (IDK might not be necessary if the schema didn't even get initialized.)

Reply:
The schema won't initialize so it should just work after they upgrade.